aea/cli/ipfs_hash.py

```diff
@@ -0,0 +1,491 @@
+#
+# ------------------------------------------------------------------------------
+
+"""
+This script generates the IPFS hashes for all packages.
+
+This script requires that you have IPFS installed:
+- https://docs.ipfs.io/guides/guides/install/
+"""
+import collections
+import csv
+import operator
+import os
+import re
+import sys
+import traceback
+from pathlib import Path
+from typing import Collection, Dict, List, Optional, Tuple, Type, cast
+
+import click
+
+from aea.configurations.base import (
+    AgentConfig,
+    ConnectionConfig,
+    ContractConfig,
+    PackageConfiguration,
+    PackageType,
+    ProtocolConfig,
+    SkillConfig,
+    _compute_fingerprint,
+)
+from aea.configurations.loader import ConfigLoaders
+from aea.helpers.ipfs.base import IPFSHashOnly
+from aea.helpers.yaml_utils import yaml_dump, yaml_dump_all
+
+
+HASHES_FILE = "hashes.csv"
+HASH_TOOL = IPFSHashOnly()
+
+type_to_class_config = {
+    PackageType.AGENT: AgentConfig,
+    PackageType.PROTOCOL: ProtocolConfig,
+    PackageType.CONNECTION: ConnectionConfig,
+    PackageType.SKILL: SkillConfig,
+    PackageType.CONTRACT: ContractConfig,
+}  # type: Dict[PackageType, Type[PackageConfiguration]]
+
+
+def package_type_and_path(package_path: Path) -> Tuple[PackageType, Path]:
+    """Extract the package type from the path."""
+    item_type_plural = package_path.parent.name
+    item_type_singular = item_type_plural[:-1]
+    return PackageType(item_type_singular), package_path
+
+
+def _get_core_packages(root_dir: Path) -> List[Tuple[PackageType, Path]]:
+    return list(
+        map(
+            package_type_and_path,
+            [
+                root_dir / "aea" / "protocols" / "scaffold",
+                root_dir / "aea" / "connections" / "scaffold",
+                root_dir / "aea" / "contracts" / "scaffold",
+                root_dir / "aea" / "skills" / "scaffold",
+            ],
+        )
+    )
+
+
+def _get_test_packages(test_data: Path) -> List[Tuple[PackageType, Path]]:
+    """Returns a list containing path for test packages."""
+    return [
+        (PackageType.AGENT, test_data / "dummy_aea"),
+        (PackageType.CONNECTION, test_data / "dummy_connection"),
+        (PackageType.CONTRACT, test_data / "dummy_contract"),
+        (PackageType.PROTOCOL, test_data / "generator" / "t_protocol"),
+        (PackageType.PROTOCOL, test_data / "generator" / "t_protocol_no_ct"),
+        (PackageType.SKILL, test_data / "dependencies_skill"),
+        (PackageType.SKILL, test_data / "exception_skill"),
+        (PackageType.SKILL, test_data / "dummy_skill"),
+    ]
+
+
+def _get_all_packages(packages_dir: Path) -> List[Tuple[PackageType, Path]]:
+    """Get all the hashable package of the repository."""
+
+    return list(
+        map(
+            package_type_and_path,
+            filter(operator.methodcaller("is_dir"), packages_dir.glob("*/*/*/")),
+        )
+    )
+
+
+def sort_configuration_file(config: PackageConfiguration) -> None:
```

Comment:
Pretty sure this already exists elsewhere, eg where we check hash consistency on startup...

Reply:
@angrybayblade

Reply:
I checked, I didn't find any method similar to this.